slybot/slybot/plugins/scrapely_annotations/extraction.py

```diff
@@ -96,6 +108,43 @@ def descriptor(self, descriptor_name=None):
         return self.descriptors.get(descriptor_name, {})
 
 
+class BaseExtractor(BasicTypeExtractor):
+    def __init__(self, annotation, attribute_descriptors=None):
+        self.annotation = annotation
+        if annotation.surrounds_attribute:
+            self.content_validate = lambda x: x
+            self.extract = self._extract_content
+
+        if annotation.tag_attributes:
+            self.tag_data = []
+            for (tag_attr, extraction_attr) in annotation.tag_attributes:
+                self.tag_data.append((lambda x: x, tag_attr, extraction_attr))
+            self.extract = self._extract_both if \
+                annotation.surrounds_attribute else self._extract_attribute
+
+    @classmethod
+    def _create_basic_extractor(cls, annotation, attribute_descriptors):
+        return cls(annotation, attribute_descriptors)
+
+class SlybotRecordExtractor(RecordExtractor):
+    def extract(self, page, start_index=0, end_index=None, ignored_regions=None, **kwargs):
+        """extract data from an extraction page
+
+        The region in the page to be extracted from may be specified using
+        start_index and end_index
+        """
+        if ignored_regions is None:
+            ignored_regions = []
+        extractors = sorted(self.extractors + ignored_regions,
+                            key=lambda x: labelled_element(x).start_index)
+        _, _, attributes = self._doextract(page, extractors, start_index,
+                                           end_index, **kwargs)
+        # collect variant data, maintaining the order of variants
+        items = []
+        for k, v in attributes:
```

Comment:
The `k, v` identifiers give the impression that `attributes` is a dict, in which case the correct way of iteration would be:
```
for k, v in attributes.iteritems()
```

If that is not the case and `attributes`is a list or other iterable then the iteration is not needed and should just simply do `items = list(attributes)`